h/app.py

```diff
@@ -46,6 +47,32 @@ def create_app(global_config, **settings):
     return config.make_wsgi_app()
 
 
+def publish_after_commit_events(request, response):
+    if request.exception is not None:
+        return
+
+    if len(request.after_commit_events) == 0:
```

Comment:
I would like to return early here so we don't have to start a new database transaction, most requests won't have any after-commit events, so we would just waste some extra round-trips to the database server.
I've changed it to `if not request.after_commit_events:` though.